src/app/data_harvester/temperature/heim.rs

```diff
@@ -89,7 +89,51 @@ pub async fn get_temperature_data(
             let temp_label = path.join(name.replace("input", "label"));
             let temp_label = fs::read_to_string(temp_label).ok();
 
-            let name = match (&hwmon_name, &temp_label) {
+            // Do some messing around to get a more sensible name for sensors
+            //
+            // - For GPUs, this will use the kernel device name, ex `card0`
+            // - For nvme drives, this will also use the kernel name, ex `nvme0`.
+            //   This is found differently than for GPUs
+            // - For whatever acpitz is, on my machine this is now `thermal_zone0`.
+            // - For k10temp, this will still be k10temp, but it has to be handled special.
+            let human_hwmon_name = {
+                let device = path.join("device");
+                // This will exist for GPUs but not others, this is how
+                // we find their kernel name
+                let drm = device.join("drm");
+                if drm.exists() {
+                    // This should never actually be empty
+                    let mut gpu = String::new();
+                    for card in drm.read_dir()? {
+                        let card = card?;
+                        let name = card.file_name().to_str().unwrap_or_default().to_owned();
+                        if name.starts_with("card") {
+                            gpu = name;
+                            break;
+                        }
+                    }
+                    Some(gpu)
```

Comment:
This might be a bit more reflective of what we're trying to do.

```rust
let mut gpu = None;
for card in drm.read_dir()? {
    let card = card?;
    let name = card.file_name().to_str().unwrap_or_default().to_owned();
    if name.starts_with("card") {
        gpu = Some(name);
        break;
    }
}
gpu
```

Reply:
Added